lib/common/client/base.rb

```diff
@@ -67,6 +68,14 @@ def request(method, path, params = {}, headers = {})
         raise client_error
       end
 
+      def sanitize_headers!(headers)
+        headers.transform_keys!(&:to_s)
+
+        headers.transform_values! do |value|
+          value.nil? ? '' : value
+        end
+      end
```

Comment:
Could this be simplified using map?

```
def sanitize_headers!(headers)
  headers.map do |k, v|
    [k.to_s, v.nil? ? '' : v]
  end
end
```

Reply:
After talking this through with harry, this would require other changes to support the non-mutating nature of map.